File: app/src/main/java/c/m/storyapp/authentication_check/domain/use_case/save_token_to_data_store_use_case/SaveTokenToDataStoreUseCase.kt
package c.m.storyapp.authentication_check.domain.use_case.save_token_to_data_store_use_case

import c.m.storyapp.authentication_check.domain.repository.AuthenticationCheckRepository
import c.m.storyapp.common.util.SimpleResource
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject

class SaveTokenToDataStoreUseCase @Inject constructor(private val authenticationCheckRepository: AuthenticationCheckRepository) {
    suspend operator fun invoke(token: String): Flow<SimpleResource> {
        return authenticationCheckRepository.saveToken(token)
    }
}

File: lib/authentication_check/domain/use_case/save_token_to_data_store_use_case/save_token_to_data_store_use_case.dart
import 'package:injectable/injectable.dart';
import 'package:story_app/authentication_check/domain/repository/authentication_check_repository.dart';

@injectable
class SaveTokenToDataStoreUseCase {
  final AuthenticationCheckRepository _authenticationCheckRepository;

  SaveTokenToDataStoreUseCase(this._authenticationCheckRepository);

  Future<void> call(String token) async {
    await _authenticationCheckRepository.saveToken(token);
  }
}
